apps/web/app/api/v1/management/responses/[responseId]/route.ts

```diff
@@ -0,0 +1,102 @@
+import { responses } from "@/lib/api/response";
+import { DatabaseError, InvalidInputError, ResourceNotFoundError } from "@formbricks/types/v1/errors";
+import { NextResponse } from "next/server";
+import { transformErrorToDetails } from "@/lib/api/validator";
+import { deleteResponse, getResponse, updateResponse } from "@formbricks/lib/services/response";
+import { TResponse, ZResponseUpdateInput } from "@formbricks/types/v1/responses";
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { getSurvey } from "@formbricks/lib/services/survey";
+import { authenticateRequest } from "@/app/api/v1/auth";
+
+async function fetchAndValidateResponse(authentication: any, responseId: string): Promise<TResponse> {
+  const response = await getResponse(responseId);
+  if (!response || !(await canUserAccessResponse(authentication, response))) {
+    throw new Error("Unauthorized");
+  }
+  return response;
+}
+
+const canUserAccessResponse = async (authentication: any, response: TResponse): Promise<boolean> => {
+  if (!authentication) return false;
+
+  const survey = await getSurvey(response.surveyId);
+  if (!survey) return false;
+
+  if (authentication.type === "session") {
+    return await hasUserEnvironmentAccess(authentication.session.user, survey.environmentId);
+  } else if (authentication.type === "apiKey") {
+    return survey.environmentId === authentication.environmentId;
+  } else {
+    throw Error("Unknown authentication type");
+  }
+};
+
+export async function GET(
+  request: Request,
+  { params }: { params: { responseId: string } }
+): Promise<NextResponse> {
+  try {
+    const authentication = await authenticateRequest(request);
+    await fetchAndValidateResponse(authentication, params.responseId);
```

Comment:
shouldn't this return the `response` variable to use later, instead of fetching again in the next line?